modules/environment/default.nix

```diff
@@ -175,6 +175,16 @@ in
 
   config = {
 
+    # This is horrible, sorry.
+    system.requiresPrimaryUser = mkIf (
+      config.nix.enable
+      && !config.nixpkgs.flake.setNixPath
+      && config.system.stateVersion < 6
+      && options.environment.darwinConfig.highestPrio == (mkOptionDefault {}).priority
```

Comment:
I hate that I can't think of a better way to check that this option wasn't changed from the default

Reply:
This is the only way I've done it, too.

Reply:
Yeah it’s awful and I hate it. But I hate not giving good error messages when doing complicated migrations even more.